public/app/features/explore/Wrapper.tsx

```diff
@@ -6,19 +6,26 @@ import { StoreState } from 'app/types';
 import { ExploreId } from 'app/types/explore';
 
 import { CustomScrollbar, ErrorBoundaryAlert } from '@grafana/ui';
-import { resetExploreAction } from './state/main';
+import { resetExploreAction, richHistoryUpdatedAction } from './state/main';
 import Explore from './Explore';
+import { getRichHistory } from '../../core/utils/richHistory';
 
 interface WrapperProps {
   split: boolean;
   resetExploreAction: typeof resetExploreAction;
+  richHistoryUpdatedAction: typeof richHistoryUpdatedAction;
 }
 
 export class Wrapper extends Component<WrapperProps> {
   componentWillUnmount() {
     this.props.resetExploreAction({});
   }
 
+  componentDidMount() {
+    const richHistory = getRichHistory();
+    this.props.richHistoryUpdatedAction({ richHistory });
```

Comment:
@ivanahuckova moved this init here. I assume it did not need to be initialised for each explore pane right?